train.py

```diff
@@ -0,0 +1,356 @@
+            label_emb = conditioning_model(velocity_bin, dstart_bin, duration_bin)
+    else:
+        label_emb = None
+
+    # get predicted noise
+    predicted_noise = model(x_noisy, t, label_emb)
+
+    # get loss value for batch
+    loss = F.mse_loss(predicted_noise, added_noise)
+
+    return loss
+
+
+@torch.no_grad()
+def validation_epoch(
+    model: Unet,
+    forward_diffusion: ForwardDiffusion,
+    conditioning_model: nn.Module,
+    dataloader: DataLoader,
+    device: torch.device,
+) -> dict:
+    # val epoch
+    val_loop = tqdm(enumerate(dataloader), total=len(dataloader), leave=False)
+    loss_epoch = 0.0
+
+    for batch_idx, batch in val_loop:
+        # metrics returns loss and additional metrics if specified in step function
+        loss = forward_step(model, forward_diffusion, conditioning_model, batch, device)
+
+        val_loop.set_postfix(loss=loss.item())
+
+        loss_epoch += loss.item()
+
+    metrics = {"loss_epoch": loss_epoch / len(dataloader)}
+    return metrics
+
+
+def save_checkpoint(
+    model: Unet, ema_model: Unet, forward_diffusion: ForwardDiffusion, optimizer: optim.Optimizer, cfg: OmegaConf, save_path: str
+):
+    # saving models
+    torch.save(
+        {
+            "model": model.state_dict(),
+            "ema_model": ema_model.state_dict(),
+            "forward_diffusion": forward_diffusion.state_dict(),
+            "optimizer": optimizer.state_dict(),
+            "config": cfg,
+        },
+        f=save_path,
+    )
+
+
+def upload_to_huggingface(ckpt_save_path: str, cfg: OmegaConf):
+    # get huggingface token from environment variables
+    token = os.environ["HUGGINGFACE_TOKEN"]
+
+    # upload model to hugging face
+    upload_file(ckpt_save_path, path_in_repo=f"{cfg.logger.run_name}.ckpt", repo_id=cfg.paths.hf_repo_id, token=token)
+
+
+@hydra.main(config_path="configs", config_name="config-default", version_base="1.3.2")
+def train(cfg: OmegaConf):
+    wandb.login()
+
+    # create dir if they don't exist
+    makedir_if_not_exists(cfg.paths.log_dir)
+    makedir_if_not_exists(cfg.paths.save_ckpt_dir)
+
+    # dataset
+    train_dataloader, val_dataloader, _ = preprocess_dataset(
+        dataset_name=cfg.train.dataset_name,
+        batch_size=cfg.train.batch_size,
+        num_workers=cfg.train.num_workers,
+        overfit_single_batch=cfg.train.overfit_single_batch,
+    )
+
+    # validate on quantized maestro
+    _, maestro_test, _ = preprocess_dataset(
+        dataset_name="JasiekKaczmarczyk/maestro-sustain-quantized",
+        batch_size=cfg.train.batch_size,
+        num_workers=cfg.train.num_workers,
+        overfit_single_batch=cfg.train.overfit_single_batch,
+    )
+
+    # logger
+    wandb.init(project="midi-diffusion", name=cfg.logger.run_name, dir=cfg.paths.log_dir)
+
+    device = torch.device(cfg.train.device)
+
+    # get trained conditioning model
+    if cfg.paths.cond_model_repo_id is not None and cfg.paths.cond_model_ckpt_path is not None:
```

Comment:
```suggestion
```

Looks like this if is only adding indentation - there's no else path, and if the condition is not true the code will break